### PR TITLE
fix: CalDAV, Cannot read properties of undefined (reading 'data')

### LINKED ISSUE #8684: Baïkal CalDAV Calendar only working one way.
Found a bug? Please fill out the sections below. 👍

### Issue Summary

In my self-hosted version of cal.com i am using a Baïkal CalDAV calendar. New bookings are written to this calendar. But it seems as cal.com does not receive any existing appointments from the calendar.

### Steps to Reproduce

Connect a Baïkal (https://sabre.io/baikal/) CalDAV calendar to cal.com 

### Technical details

I am using the version of May 4, 2023.

The node version is v16.19.1

I get this output:
```
@calcom/web:start: TypeError: Cannot read properties of undefined (reading 'data')
@calcom/web:start:     at /root/cal.com/apps/web/.next/server/chunks/1012.js:273:24
@calcom/web:start:     at Array.forEach (<anonymous>)
@calcom/web:start:     at CalDavCalendarService.getAvailability (/root/cal.com/apps/web/.next/server/chunks/1012.js:272:17)
@calcom/web:start:     at processTicksAndRejections (node:internal/process/task_queues:96:5)
@calcom/web:start:     at async /root/cal.com/apps/web/.next/server/chunks/39906.js:191:32
@calcom/web:start:     at async Promise.all (index 0)
@calcom/web:start:     at async getCachedResults (/root/cal.com/apps/web/.next/server/chunks/39906.js:199:28)
@calcom/web:start:     at async getStaticProps (/root/cal.com/apps/web/.next/server/pages/[user]/calendar-cache/[month].js:56:60)
@calcom/web:start:     at async Object.renderToHTML (/root/cal.com/node_modules/next/dist/server/render.js:386:20)
@calcom/web:start:     at async doRender (/root/cal.com/node_modules/next/dist/server/base-server.js:809:34)
@calcom/web:start:     at async cacheEntry1.responseCache.get.incrementalCache.incrementalCache (/root/cal.com/node_modules/next/dist/server/base-server.js:929:28)
@calcom/web:start:     at async /root/cal.com/node_modules/next/dist/server/response-cache/index.js:83:36 Cannot read properties of undefined (reading 'data')
@calcom/web:start: Error setting header x-csp=not-opted-in for /sales/calendar-cache/2023-05 Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
@calcom/web:start:     at new NodeError (node:internal/errors:387:5)
@calcom/web:start:     at ServerResponse.setHeader (node:_http_outgoing:644:11)
@calcom/web:start:     at ServerResponse._res.setHeader (/root/cal.com/node_modules/next/dist/server/base-server.js:180:24)
@calcom/web:start:     at setHeader (/root/cal.com/apps/web/.next/server/pages/_document.js:118:18)
@calcom/web:start:     at Function.getInitialProps (/root/cal.com/apps/web/.next/server/pages/_document.js:128:13)
@calcom/web:start:     at Object.<anonymous> (/root/cal.com/node_modules/next/dist/shared/lib/utils.js:84:33)
@calcom/web:start:     at Generator.next (<anonymous>)
@calcom/web:start:     at asyncGeneratorStep (/root/cal.com/node_modules/@swc/helpers/lib/_async_to_generator.js:13:28)
@calcom/web:start:     at _next (/root/cal.com/node_modules/@swc/helpers/lib/_async_to_generator.js:31:17)
@calcom/web:start:     at /root/cal.com/node_modules/@swc/helpers/lib/_async_to_generator.js:36:13
@calcom/web:start:     at new Promise (<anonymous>)
@calcom/web:start:     at Object.<anonymous> (/root/cal.com/node_modules/@swc/helpers/lib/_async_to_generator.js:28:16)
@calcom/web:start:     at Object.loadGetInitialProps (/root/cal.com/node_modules/next/dist/shared/lib/utils.js:62:33)
@calcom/web:start:     at loadDocumentInitialProps (/root/cal.com/node_modules/next/dist/server/render.js:682:48)
@calcom/web:start:     at renderDocument (/root/cal.com/node_modules/next/dist/server/render.js:732:45)
@calcom/web:start:     at /root/cal.com/node_modules/next/dist/server/render.js:769:114 {
@calcom/web:start:   code: 'ERR_HTTP_HEADERS_SENT'
@calcom/web:start: }
```

### PR DESCRIPTION
**Pull request by alishaz-polymath**
## What does this PR do?

Handles the object.data check better to first confirm if object itself is not undefined

Fixes #8684 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [x] When the calendar returns undefined object, it should simply ignore it instead of throwing the error `Cannot read properties of undefined (reading 'data')` in the console.

## Mandatory Tasks
- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/lib/CalendarService.ts [hunk 1]
```diff
@@ -337,7 +337,7 @@ export default abstract class BaseCalendarService implements Calendar {
     const userTimeZone = userId ? await this.getUserTimezoneFromDB(userId) : "Europe/London";
     const events: { start: string; end: string }[] = [];
     objects.forEach((object) => {
-      if (object.data == null || JSON.stringify(object.data) == "{}") return;
+      if (!object || object.data == null || JSON.stringify(object.data) == "{}") return;
```
Comment (alishaz-polymath) [inline packages/lib/CalendarService.ts:340]: Checks if object exists before trying to access properties within.

**Review by alishaz-polymath — COMMENTED** [review]
Comment

File: packages/lib/CalendarService.ts (hunk 1 above)
Comment (vsviridov) [reply packages/lib/CalendarService.ts:340]: Proper types should've prevented this.

Comment (emrysal) [reply packages/lib/CalendarService.ts:340]: Agreed, we'd have had to type these ourselves and that would have been error prone too - types do not prevent this.

**Review by emrysal — APPROVED** [review]

**Pull request merged**